capstone-app/src/app/modules/photos/components/image-container/image-container.component.spec.ts

```diff
@@ -19,7 +53,37 @@ describe('ImageContainerComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  describe('when test with asynchronous observable', () => {
+    beforeEach(() => {
+      getPhotosSpy.and.returnValue(asyncData(testPhotos));
+    });
+
+    it('should display a list of photos', () => {
+      const bannerDe: DebugElement = fixture.debugElement;
+      const bannerEl: HTMLElement = bannerDe.nativeElement;
+      const imageList = bannerEl.querySelectorAll('img');
+      const testPhotosLinks: string[] = [];
+      const imageListLinks: string[] = [];
+      for (const image in imageList) {
```

Comment:
Optional: Consider using the "functional" way of doing this:

```
const imageListLinks = imageList
    // Removes images with no src
    .filter(image => !!image.src)
    .map(image => `..${image.src.substring(21, image.src.length)}`);
```

Also: You almost NEVER want a "for...in" loop, because that will only give you the indices. A "for...of" loop is preferred because then you can just have access to "image" itself instead of doing "imageList[image]" everywhere.

Reply:
imageList is a NodeListOf that doesn't have an iterator type, unfortunately.

Reply:
Converted to array to get around that.